dell/src/main/java/org/apache/iceberg/dell/ObjectKey.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.dell;
+
+import java.util.Objects;
+
+/**
+ * a immutable record class of object key
+ */
+public class ObjectKey {
+  /**
+   * bucket
+   */
+  private final String bucket;
+
+  /**
+   * key
+   */
+  private final String key;
+
+  public ObjectKey(String bucket, String key) {
```

Comment:
Something is not right here. The class is called `ObjectKey`, but it is created by passing a `String` called `key`. Is the string the key, or is it part of the key? Maybe this should be similar to S3, where the bucket/string combination is called a URI, `S3URI`.

Reply:
In object storage, the object key contains both a bucket name and a key.
And in my design, the user can provide different delimiters for object storage.
For example, the user can create an object which bucket is _a_ and key is _folder2021-folder09-obj1_ with delimiter _-_.
It's hard to format it as a string, so I move this to {ObjectKeys} interface.

And I need some time to check whether using `emc://bucket/key` as a general string.